### PR TITLE
Replace JSCS/JSHint with ESLint & lint by editorconfig

### PR DESCRIPTION
**Pull request by shvaikalesh**
Equality changes should not break stuff: same types are being compared.
Also, renamed `validate` to `lint` in `"scripts"`.

### CODE REVIEW

**Comment by pepelsbey** [comment]
There’s also [editorconfig-cli](https://github.com/htmlacademy/editorconfig-cli) package worth adding. It could test all existing files againts `.editorconfig`.

File: Gulpfile.js [hunk 1]
```diff
@@ -19,7 +20,29 @@ const banner = `/**
  */
 `;
 
-gulp.task('lint', () => {
+gulp.task('lint:ec', () => {
+    const sources = [
```
Comment (shvaikalesh) [inline Gulpfile.js]: Because we have to check dotfiles too, and ignore built ones and `node_modules`, it is more convenient to whitelist than blacklist.

**Comment by shvaikalesh** [comment]
I had concerns with level of support and code quality of proposed solution and used `gulp-lintspaces` instead.

**Comment by pepelsbey** [comment]
I guess only `lint-staged` is left. Right?

**Comment by shvaikalesh** [comment]
I planned to refresh git hooks in the next PR.

**Comment by pepelsbey** [comment]
:shipit: then :)

**Pull request merged**